service/src/main/java/com/epam/esm/logic/impl/GiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.esm.logic.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.exception.DuplicateEntityException;
+import com.epam.esm.exception.NotFoundEntityException;
+import com.epam.esm.logic.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class GiftCertificateServiceImpl implements GiftCertificateService {
+    private final GiftCertificateDao giftCertificateDao;
+    private final TagDao tagDao;
+
+    @Autowired
+    public GiftCertificateServiceImpl(GiftCertificateDao giftCertificateDao, TagDao tagDao) {
+        this.giftCertificateDao = giftCertificateDao;
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    @Transactional
+    public long create(GiftCertificateDto giftCertificateDto) {
+        GiftCertificate giftCertificate = giftCertificateDto.getGiftCertificate();
+        //todo validation
+        if (giftCertificateDao.findByName(giftCertificate.getName()).isPresent()) {
+            throw new DuplicateEntityException("Certificate already exist");
+        }
+        giftCertificateDao.create(giftCertificate);
+        return -1L;
+    }
+
+    @Override
+    public List<GiftCertificate> findAll() {
+        return giftCertificateDao.findAll();
+    }
+
+    @Override
+    public GiftCertificate findById(long id) {
+        Optional<GiftCertificate> giftCertificateOptional = giftCertificateDao.findById(id);
+        if (!giftCertificateOptional.isPresent()) {
```

Comment:
Don't start checking a condition with negation. Move to a separate private method.